core/state/commonstate/deprecated_state.go

```diff
@@ -0,0 +1,175 @@
+package commonstate
+
+import (
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/state/commontrie"
+)
+
+// DeprecatedStateAdapter wraps core.State to implement CommonState
+type DeprecatedStateAdapter core.State
```

Comment:
Instead of this, I'm wondering if we can refactor the old state to return `felt.Felt` instead of `*felt.Felt`, so we don't need this adapter.

Reply:
Refactored the state anyways